### PR TITLE
Implement multipart upload for gcloud.aio.storage

### PR DESCRIPTION
**Pull request by AlexisH**
Support for uploading a file and settings metadata in a single request

### CODE REVIEW

**Comment by AlexisH** [comment]
Requests is dependent on urllib3 so there is no added dependency at least
for gcloud-rest

Le jeu. 30 juil. 2020 à 02:02, Kevin James <notifications@github.com> a
écrit :

**Comment by AlexisH** [comment]
for gcloud-aio we could implement another version using the
aiohttp.multipart package. https://docs.aiohttp.org/en/stable/multipart.html


Alexis

File: storage/gcloud/aio/storage/storage.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from typing import Tuple
 from typing import Union
 from urllib.parse import quote
+from urllib3.filepost import encode_multipart_formdata, choose_boundary
+from urllib3.fields import RequestField
```
[the remark this answers is not in the document]
Comment (AlexisH) [reply storage/gcloud/aio/storage/storage.py:14]: Requests is dependent on urllib3 so there is no added dependency for gcloud-rest-*
For gcloud-aio we could implement another version using the
[aiohttp.multipart](https://docs.aiohttp.org/en/stable/multipart.html) package.

**Comment by TheKevJames** [comment]
Fixed up and merged as 2bc3b4bf30eb3b9f93fd057339ecfe5b191bf4d5. Will be included in `gcloud-*-storage==5.7.0`

**Pull request closed without merging**